pandas/core/indexes/base.py

```diff
@@ -4916,7 +4928,13 @@ def _get_string_slice(self, key: str_t, use_lhs: bool = True, use_rhs: bool = Tr
         # overridden in DatetimeIndex, TimedeltaIndex and PeriodIndex
         raise NotImplementedError
 
-    def slice_indexer(self, start=None, end=None, step=None, kind=None):
+    def slice_indexer(
+        self,
+        start: Optional[Label] = None,
+        end: Optional[Label] = None,
```

Comment:
isnt Optional redundant here?

Reply:
we've got `no_implicit_optional=True` in out setup.cfg (but I don't know why?)

it would work, but only since we have Label = Optional[Hashable] to allow `None` to be a Label. Label would be easier to grok if it were Union[Hashable, None] (or if Hashable included None) but for consistency we use Optional.

Here, for consistency of keyword parameter annotations, don't really want to remove the Optional, even though already accounted for in Label.

but I'll change if blocker.